internal/interpreter/dynamic_item_tree.rs

```diff
@@ -1420,8 +1420,16 @@ pub fn animation_for_property(
                     let state = eval::eval_expression(&state_ref, &mut context);
                     let state_info: i_slint_core::properties::StateInfo = state.try_into().unwrap();
                     for a in &animations {
-                        if (a.is_out && a.state_id == state_info.previous_state)
-                            || (!a.is_out && a.state_id == state_info.current_state)
+                        if (a.direction
+                            == i_slint_compiler::object_tree::TransitionDirection::Input
+                            && a.state_id == state_info.previous_state)
+                            || (a.direction
+                                == i_slint_compiler::object_tree::TransitionDirection::Output
+                                && a.state_id == state_info.current_state)
+                            || (a.direction
+                                == i_slint_compiler::object_tree::TransitionDirection::InOut
+                                && (a.state_id == state_info.current_state
+                                    || a.state_id == state_info.previous_state))
```

Comment:
perhaps this code would be simplified with a `match`

Reply:
How about the latest code?